packages/opentelemetry-plugin-grpc/src/grpc.ts

```diff
@@ -0,0 +1,625 @@
+      'removing patch to %s@%s',
+      this.moduleName,
+      this.version
+    );
+
+    if (this._moduleExports.Server) {
+      shimmer.unwrap(this._moduleExports.Server.prototype, 'register');
+    }
+
+    if (grpcClientModule) {
+      shimmer.unwrap(grpcClientModule, 'makeClientConstructor');
+    }
+  }
+
+  private static _toSpanStatus(status: number): Status {
+    return { code: status };
+  }
+
+  private _getSpanContext(metadata: grpcModule.Metadata): SpanContext | null {
+    const metadataValue = metadata.getMap()[GRPC_TRACE_KEY] as Buffer;
+    // Entry doesn't exist
+    if (!metadataValue) {
+      return null;
+    }
+    const spanContext = this._tracer.getBinaryFormat().fromBytes(metadataValue);
+    // Value is malformed
+    if (!spanContext) {
+      return null;
+    }
+    return spanContext;
+  }
+
+  private _setSpanContext(
+    metadata: grpcModule.Metadata,
+    spanContext: SpanContext
+  ): void {
+    const serializedSpanContext = this._tracer
+      .getBinaryFormat()
+      .toBytes(spanContext);
+    const buffer = Buffer.from(serializedSpanContext);
+    metadata.set(GRPC_TRACE_KEY, buffer);
+  }
+
+  private _patchServer() {
+    return (originalRegister: typeof grpcModule.Server.prototype.register) => {
+      const plugin = this;
+      plugin._logger.debug('patched gRPC server');
+
+      return function register<RequestType, ResponseType>(
+        // tslint:disable-next-line:no-any
+        this: grpcModule.Server & { handlers: any },
+        name: string,
+        handler: grpcModule.handleCall<RequestType, ResponseType>,
+        serialize: grpcModule.serialize<RequestType>,
+        deserialize: grpcModule.deserialize<RequestType>,
+        type: string
+      ) {
+        // tslint:disable-next-line:no-any
+        const originalResult = originalRegister.apply(this, arguments as any);
+        const handlerSet = this.handlers[name];
+
+        shimmer.wrap(
+          handlerSet,
+          'func',
+          (originalFunc: grpcModule.handleCall<RequestType, ResponseType>) => {
+            return function func(
+              this: typeof handlerSet,
+              call: ServerCallWithMeta,
+              callback: SendUnaryDataCallback
+            ) {
+              const self = this;
+
+              const spanName = `grpc.${name.replace('/', '')}`;
+              const spanOptions: SpanOptions = {
+                kind: SpanKind.SERVER,
+              };
+
+              const parentSpan = plugin._getSpanContext(call.metadata);
+              if (parentSpan) {
+                spanOptions.parent = parentSpan;
+              }
+              plugin._logger.debug(
+                'patch func: %s',
+                JSON.stringify(spanOptions)
+              );
+
+              const span = plugin._tracer
+                .startSpan(spanName, spanOptions)
+                .setAttribute(AttributeNames.COMPONENT, GrpcPlugin.component);
+
+              if (spanOptions.kind) {
```

Comment:
This can be combined with above, something like:
```
.setAttributes({
  AttributeNames.COMPONENT, GrpcPlugin.component, 
  GrpcPlugin.ATTRIBUTE_GRPC_KIND: SpanKind.SERVER // kind = SpanKind.SERVER, for server patch. 
});
```